src/components/P/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import styled from "styled-components";
+import PropTypes from "prop-types";
+
+interface Props {
+  fontSize?: string;
+  fontWeight?: string;
+  color?: string;
+  textAlign?: string;
+  marginLeft?: string;
+  marginBottom?: string;
+  marginTop?: string;
+  width?: string;
+  shadow?: string;
+}
```

Comment:
Interesting fact, in case you need to have an interface where all those properties are required and another one where all are optional you could write it this way:
```
interface Props {
  fontSize: string;
  fontWeight: string;
  color: string;
  textAlign: string;
  marginLeft: string;
  marginBottom: string;
  marginTop: string;
  width: string;
  shadow: string;
}

interface PropsOptional extends Partial<Props>
```

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html